src/components/CardContainer/CardContainer.js

```diff
@@ -30,12 +30,28 @@ class CardContainer extends React.Component {
             .then(res => {
                 let random = UniqueRandomArray(res)
                 let dogs = [random(), random()]
-                const state = {...this.state, dogs: dogs}
+                const state = {...this.state, dogs: dogs, randomDog: random}
                 this.setState(state)
                 }
             )
     }
 
+    clickUpdateWin = (id) => {
+        fetch('http://localhost:3000/dogs/' + id + "/win", {
+            method: 'POST'
+        })
+    }
+
+    refreshDogs = () => {
+        let dogs = [this.state.randomDog(), this.state.randomDog()]
+        while ((dogs[0]._id == this.state.dogs[0]._id || dogs[0]._id == this.state.dogs[1]._id)
```

Comment:
So, at the moment if there's a problem and we can't get two different dogs then we will keep this while loop going forever.
I wonder if we could use a for-loop instead, and then we could limit how many tries we want to do before maybe giving up, displaying the two same dogs and alerting the user that there has been a problem? 
Also I think an `OR` should be fine between the two dogs check - 
`WHILE - dog 1 has changed OR dog 2 has changed` but the logic is complicated to happy to be challenged on that

Reply:
on it

Reply:
Another thought - maybe we could pull the `haveDogsChanged` aspect out into another function to split out the logic a bit

Reply:
however this is just a suggestion - maybe try it out and see how it looks

Reply:
Has anyone looked at this?